create-seaweedfs/main.tf

```diff
@@ -0,0 +1,68 @@
+locals {
+  chart_values = {
+    "POSTGRESQL_HOST"     = var.postgresql_host
+    "POSTGRESQL_PORT"     = var.postgresql_port
+    "POSTGRESQL_DATABASE" = var.postgresql_database
+    "POSTGRESQL_USERNAME" = var.postgresql_username
+    "POSTGRESQL_SECRET"   = var.postgresql_password_secret
+    "FILER_ENDPOINT"      = "http://${local.release_name}-filer.${var.namespace}.svc.cluster.local:8888"
+    "S3_AUTH_SECRET"      = "${local.release_name}-s3-config"
+    "S3_INIT_BUCKETS"     = ["argo-workflows"]
+  }
+  s3_config_values = {
+    "ARGO_WORKFLOWS_USERNAME" = local.argo_workflows_username
+    "ARGO_WORKFLOWS_PASSWORD" = random_password.argo_workflows_password.result
+  }
+  release_name            = "seaweedfs-${var.namespace}"
+  argo_workflows_username = "argo_workflows"
+}
+
+resource "helm_release" "seaweedfs" {
+  name       = local.release_name
+  repository = "https://charts.bitnami.com/bitnami"
+  chart      = "seaweedfs"
+  version    = "3.0.3"
```

Comment:
I know we have variables elsewhere, but for me it bring nothing more. The variables are not exposed outside the main terraform volume. So the internal module variable are basically just local variable. You end up with a fixed hardcoded version anyway. As long as the version value is not duplicated I prefer to avoid useless internal variables. Same with the chart name or repository.

Reply:
but...If you leave the variable fixed, you won't be able to upgrade this chart without modifying the module afterwards (so new tag then new release), because the module is called from terraform registry.

Reply:
I've exposed the seaweedfs chart version at the module root.